### PR TITLE
proxmox inventory - fix parsing for offline nodes

### LINKED ISSUE #2931: Proxmox Inventory fails when a cluster node is offline
### Summary

When querying the proxmox API with the ansible inventory plugin. If one server of the cluster is offline then the inventory fails to build.

i.e. Proxmox inventory connects to server2 and if server1 is offline then the proxmox inventory is unable to generate an inventory.

### Issue Type

Bug report

### Component Name

plugins/inventory/proxmox.py

### Ansible Version

```console (paste below)
$ ansible --version
ansible 2.10.7
  config file = /home/adamus1red/ansible/ansible.cfg
  configured module search path = ['/home/adamus1red/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/adamus1red/ansible/python-debian/lib/python3.7/site-packages/ansible
  executable location = /home/adamus1red/ansible/python-debian/bin/ansible
  python version = 3.7.3 (default, Jan 22 2021, 20:04:44) [GCC 8.3.0]
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed
ANSIBLE_SSH_ARGS(/home/adamus1red/ansible/ansible.cfg) = -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no
INVENTORY_ENABLED(/home/adamus1red/ansible/ansible.cfg) = ['host_list', 'script', 'auto', 'yaml', 'ini', 'toml', 'vmware_vm_inventory', 'community.general.proxmox']
```

### OS / Environment

Proxmox Debian

### Inventory
```yaml
plugin: community.general.proxmox
url: https://pve-02.lan:8006
user: Ansible@pve
password: ****
validate_certs: yes
want_facts: yes
```
### Playbook
```yaml
---
- hosts: all
  become: true
  remote_user: root
  roles:
    - base
```

### Expected Results

I expect an inventory to be generated from the remaining hosts in the cluster.

### Actual Results

```console (paste below)
[WARNING]:  * Failed to parse /home/adamus1red/ansible/pve.proxmox.yml with auto plugin: 595 Server Error: No route to host for url: https://pve-02.lan:8006/api2/json/nodes/pve-01/network
[WARNING]:  * Failed to parse /home/adamus1red/ansible/pve.proxmox.yml with yaml plugin: Plugin configuration YAML file, not YAML inventory
[WARNING]:  * Failed to parse /home/adamus1red/ansible/pve.proxmox.yml with ini plugin: Invalid host pattern 'plugin:' supplied, ending in ':' is not allowed, this character is reserved to provide a port.
[WARNING]:  * Failed to parse /home/adamus1red/ansible/pve.proxmox.yml with ansible_collections.community.general.plugins.inventory.proxmox plugin: 595 Server Error: No route to host for url:
https://pve-02.lan:8006/api2/json/nodes/pve-01/network
[WARNING]: Unable to parse /home/adamus1red/ansible/pve.proxmox.yml as an inventory source
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
[WARNING]: Could not match supplied host pattern, ignoring: ven-*

```

When server pve-01 is online the inventory is generated as expected.

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by Ajpantuso**
##### SUMMARY
Adding logic to skip collecting facts and guests for `offline` nodes.

Fixes #2931

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
plugins/inventory/proxmox.py

##### ADDITIONAL INFORMATION
- Removed some unused imports and variables in unit tests as well,

### CODE REVIEW

**Comment by Thulium-Drake** [comment]
LGTM

Thank you for this fix! 😁

**Comment by felixfontein** [comment]
recheck

File: changelogs/fragments/2967-proxmox_inventory-offline-node-fix.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+bugfixes:
+  - proxmox inventory - fixed parsing failures when some cluster nodes are offline (https://github.com/ansible-collections/community.general/issues/2931).
```
Comment (felixfontein) [inline changelogs/fragments/2967-proxmox_inventory-offline-node-fix.yml:3]:
```suggestion
  - proxmox inventory plugin - fixed parsing failures when some cluster nodes are offline (https://github.com/ansible-collections/community.general/issues/2931).
```

**Review by felixfontein — COMMENTED** [review]
Besides that, looks good!

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@Ajpantuso thanks for fixing this!
@Thulium-Drake thanks for reviewing this!

**Pull request merged**